tests/queries_tests/tag_test.js

```diff
@@ -0,0 +1,41 @@
+import chai, { expect } from 'chai'
+import * as tag from '../../src/database/queries/tag'
+
+describe('tag', () => {
+
+  const fakeTags = [
+    {
+      names: 'Massage'
+    },
+    {
+      names: 'Spa'
+    }
+  ]
+
+  beforeEach( () =>
+    Promise.all([
+      tag.deleteAll(),
+      tag.add(fakeTags)
+    ])
+  )
+
+  it('should exist', () =>
+    expect(tag).to.be.a('object')
+  )
+
+  it('should return all tags ordered ascending by names', () =>
+    tag.getAll().then( tags => {
+      expect( fakeTags[0].names ).to.equal('Massage')
+      expect( fakeTags[1].names ).to.equal('Spa')
+    })
```

Comment:
![](https://the900facebookpony.files.wordpress.com/2016/03/2a76a-lostinspace.gif)

This is a huge problem that happens all the time. You have accidentally made a test that doesn't test you code. 

It helps to avoid this if you remeber to always make your tests fail in predictable ways. If you write a test and is passes __be skeptical__! Go into the code your testing and make sure intended failures cause intended tests to fail. 


If you had done that you would have notice this test doesn't even look at the result of `tag.getAll()`. All you're doing is confirming your `fakeTags` array has the data you gave it. 

```js
it('should return all tags ordered ascending by names', function(){
  return tag.getAll()
    .then( tags => {
      expect( tags.length ).to.equal(fakeTags.length)
      expect( tags[0].names ).to.equal(fakeTags[0].names)
      expect( tags[1].names ).to.equal(fakeTags[0].names)
    })
})
```